src/components/GoodsList.tsx

```diff
@@ -0,0 +1,165 @@
+import React from 'react';
+import './GoodsList.scss';
+
+interface Props {
+  goods: string[];
+}
+
+interface State extends Props {
+  reverse: boolean;
+  sortBy: string;
+  selectValue: string;
+}
+
+export class GoodsList extends React.Component<Props, State> {
+  state = {
+    goods: this.props.goods,
+    reverse: false,
+    sortBy: '',
+    selectValue: '1',
+  };
+
+  sortAlphabetical = () => {
+    this.setState({
+      sortBy: 'alphabetical',
+    });
+  };
+
+  sortByLength = () => {
+    this.setState({
+      sortBy: 'length',
+    });
+  };
```

Comment:
you can create 1 sort method and set sortBy using parameters that will be passed to the function